clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -791,8 +791,17 @@ static void VisitField(CXXRecordDecl *Owner, RangeTy &&Item, QualType ItemTy,
   else if (ItemTy->isStructureOrClassType())
     VisitAccessorWrapper(Owner, Item, ItemTy->getAsCXXRecordDecl(),
                          handlers...);
+#if 0
+  // FIXME Enable this when structs are replaced by their fields
   else if (ItemTy->isArrayType())
     VisitArrayElements(Item, ItemTy, handlers...);
```

Comment:
Seems like it adds support for arrays of arrays and etc. Why It can be enabled only when structs are replaced by their fields?

Reply:
Existing tests that contain arrays within structs fail because the other elements of those structs are not decomposed yet. Once struct decomposition is enabled, this can be enabled too. Additional array tests will also pass.